### DISCUSSION TITLE
Disable filtering and sorting in tables.

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#2626] [PrimeFaces] [answered]
Is it possible to disable filtering or sorting while a filtering or sorting is in progress?

**Comment by melloware** [comment, marked as answer]
Yes see BlockUI second example: https://www.primefaces.org/showcase/ui/misc/blockUI.xhtml

**Reply by hgfrey** [reply]
Thanks. I'd like to try this out.

**Reply by hgfrey** [reply]
How can I trigger the blockUI element when sorting or filtering has been triggered?

**Reply by melloware** [reply]
According to the docs: https://primefaces.github.io/primefaces/15_0_0/#/components/blockui?id=client-side-api

There is `show()` and `hide()` widget methods.

```js
PF('blockUI').show();
```

**Reply by hgfrey** [reply]
What I mean is, how can I use blockUi so that when clicking sort or filtering, no further input is possible until the request is returned and the UI has been redrawn, i.e. the result of the sorting or filtering is displayed.

**Reply by melloware** [reply]
Did you even look at the Showcase examples?   Isn't that the second Example right here: https://www.primefaces.org/showcase/ui/misc/blockUI.xhtml  using a DataTable if you sort, page, filter it blocks it automatically and returns?

**Reply by hgfrey** [reply]
And what do I specify as a trigger? The ID of the table?

**Reply by melloware** [reply]
ummm yep just like in the SHowcase example. `trigger="dataTable"` or whatever the ID of your table is.

**Reply by hgfrey** [reply]
Okay. Thank you...

**Reply by hgfrey** [reply]
blockUI works as intended for scrolling and changing the amount of rows displayed. Unfortunately, this doesn't work for filtering and sorting. Is there a way to trigger blockUI through filtering and sorting? The problem is that people get impatient and fidget with the UI without realizing that each click is being queued and processed one by one.

**Reply by melloware** [reply]
it works I just tested it?  If I filter or sort it blocks?

**Reply by hgfrey** [reply]
strange. :-(

**Reply by melloware** [reply]
I just added an example here click on the Sort for the column and you will see the Block UI take effect. Of Filter and press ENTER

https://quarkus-faces-melloware-8a6a34c1.koyeb.app/ui/misc/blockUI.xhtml

**Reply by hgfrey** [reply]
Excuse me. It's my mistake. I'm currently seeing the following:

```
                <p:ajax event="filter" ignoreAutoUpdate="true" global="false"/>
                <p:ajax event="sort" listener="#{ptPlanungController.onSort}" global="false"/>
```
